sanctum-solana-client-utils/src/prioritization_fee.rs

```diff
@@ -0,0 +1,154 @@
+use std::cmp::min;
+
+use medians::Medianf64;
+use solana_client::{
+    nonblocking::rpc_client::RpcClient as NonblockingRpcClient, rpc_client::RpcClient,
+    rpc_response::RpcPrioritizationFee,
+};
+use solana_rpc_client_api::{
+    client_error::Error as ClientError, config::RpcSimulateTransactionConfig,
+};
+use solana_sdk::{
+    compute_budget::ComputeBudgetInstruction, instruction::Instruction, message::Message,
+    pubkey::Pubkey, transaction::Transaction,
+};
+
+const MAX_SLOT_DISPLACEMENT: u64 = 150;
+const WEIGHTED_MEDIAN_EPSILON: f64 = 0.0001;
+
+pub fn get_compute_budget_ixs(unit_limit: u32, unit_price_micro_lamports: u64) -> [Instruction; 2] {
+    [
+        ComputeBudgetInstruction::set_compute_unit_limit(unit_limit),
+        ComputeBudgetInstruction::set_compute_unit_price(unit_price_micro_lamports),
+    ]
+}
+
+// assumes <= `MAX_SLOT_DISPLACEMENT` slots of sample size
+pub fn calc_slot_weighted_median_prioritization_fees(
+    rpc_prio_fees: &[RpcPrioritizationFee],
+) -> u64 {
+    if rpc_prio_fees.is_empty() {
+        log::warn!("Given sample for the recent prioritization fee was empty");
```

Comment:
If we return `Option<>` here we can early return on `max_by_key()`, and users dont have to wonder if a returned 0 means the samples were empty or if it was actually 0